### DISCUSSION TITLE
How to test basicvsr++ on long video?

### DISCUSSION THREAD

**Discussion by Luciennnnnnn** [open-mmlab/mmediting#1208] [How-to] [answered]
@ckkelvinchan Hi, I have a simple question, how to test basicvsr++ on long video with many frames. I notice that you process whole video frames together as your config file shows. REDS dataset only has 100 frames per video clip, so this method will not have a problem, but when I want to test on a longer video, such as 300/6000 frames, how can I achieve it?

**Comment by meobach** [comment, marked as answer]
you can check the file mmedit/apis/restoration_video_inference.py, in that file, It has the variable name: max_seq_len and you can set it to value with the maximal number of frames your computer can handle. It will process each sequence with length: max_seq_len and concat all the results at the end.

**Comment by Luciennnnnnn** [comment]
@grimoire Yeah, I found this very helpful, thank you for your timely response! But, I have another question, the last split may have the length of frames smaller than `max_seq_len`, why should we use `max_seq_len` frames from the end to the beginning as input, and only use part of the result we lack, this seems more consistent.

**Comment by meobach** [comment]
sure, you can test and I think it can help improve your result. The approach to splitting is the simplest one.

**Comment by Luciennnnnnn** [comment]
@meobach Thanks!
